src/LinAlg/hiopLinSolverIndefDenseMagma.cpp

```diff
@@ -1,49 +1,116 @@
 #include "hiopLinSolverIndefDenseMagma.hpp"
 
+#include "hiopMatrixRajaDense.hpp"
+
 namespace hiop
 {
   hiopLinSolverIndefDenseMagmaBuKa::hiopLinSolverIndefDenseMagmaBuKa(int n, hiopNlpFormulation* nlp_)
-    : hiopLinSolverIndefDense(n, nlp_), work_(NULL)
+    : hiopLinSolverIndefDense(n, nlp_)
   {
-    ipiv_ = new int[n];
+    magma_int_t ndevices;
+    magma_device_t devices[MagmaMaxGPUs];
+    magma_getdevices(devices, MagmaMaxGPUs, &ndevices);
+    assert(ndevices>=1);
+
+    int device = 0;
+    magma_setdevice(device);
+
+    magma_queue_create(devices[device], &magma_device_queue_);
+
+    int magmaRet;
+
+    const int align=32;
+    ldda_ = magma_roundup(n, align );  // multiple of 'align', small power of 2 (i.e., 32)
+    
+    const int nrhs=1;
+    lddb_ = ldda_;
+
+    magmaRet = magma_malloc((void**)&dinert_, 3*sizeof(int));
+    assert(MAGMA_SUCCESS == magmaRet);
+
+
+    magmaRet = magma_imalloc_pinned(&ipiv_, ldda_);
+    assert(MAGMA_SUCCESS == magmaRet);
```

Comment:
Do you need to use pinned memory here?

Reply:
it looks like... it's not documented by magma, but in their example uses pinned memory and magma crashes with other memory allocation types..